docs/service-bundle.md

```diff
@@ -0,0 +1,225 @@
+creating a named secret (POD_NAME) in the sandbox namespace (POD_NAMESPACE).
+The secret should have a single key called “fields” whose value is a
+base64-encoded JSON document.
+
+Ansible Playbook Bundles use the “asb_encode_binding” module to do this.
+
+```
+apiVersion: v1
+data:
+  fields: eyJEQl9OQU1FIjogImZvbyIsICJEQl9QQVNTV09SRCI6ICJsZXRtZWluIiwgIkRCX1RZUEUiOiAicG9zdGdyZXMiLCAiREJfUE9SVCI6ICI1NDMyIiwgIkRCX1VTRVIiOiAiYmFyIiwgIkRCX0hPU1QiOiAicG9zdGdyZXNxbCJ9
+kind: Secret
+metadata:
+  creationTimestamp: 2018-02-26T18:34:50Z
+  name: apb-874ae78d-acfe-4181-b064-124d6e13c592
+  namespace: dh-postgresql-apb-prov-md5xt
+  resourceVersion: "11846"
+  selfLink: /api/v1/namespaces/dh-postgresql-apb-prov-md5xt/secrets/apb-874ae78d-acfe-4181-b064-124d6e13c592
+  uid: ba79e27a-1b23-11e8-aebf-94b96c9ea8ca
+type: Opaque
+```
+
+The value for “fields” is a base64-encoded JSON document whose keys and values
+represent any data the APB wants to return to the broker and ultimately to a
+client that retrieves a binding.
+
+__Provision Credentials__ - If a service bundle saves such a secret during the
+provision operation, the broker saves the passed data as “provision
+credentials”. That data gets passed into all further actions as part of the
+JSON document under the key ``_apb_provision_creds``, as a sub-document. As such,
+it can be useful for making root-level credentials created during provision
+available to other operations that need to use them. If a request for non-async
+binding is made and the bundle does not run, any “provision credentials” that
+were saved will be returned to the client as binding credentials.
+
+__Binding Credentials__ - If a service bundle saves such a secret during the bind
+operation, the broker saves the passed data as “binding credentials”. These
+credentials get returned to the client that initiated that specific binding
+operation.
+
+## Runtime
+
+The container will be run in a pod by itself with ``restartPolicy: Never``. The
+pod will run in a sandbox namespace created by the broker for that specific
+operation only. The namespace will be deleted at some point after the pod
+finishes.
+
+APB is run with a service account granted access to the ``POD_NAMESPACE`` and
+to the “namespace” that the pod is to deploy to. Networks between the two
+namespaces are tied together so that you can reliably find your services in the
+target namespace. That only works for
+[NetworkPolicy](https://kubernetes.io/docs/concepts/services-networking/network-policies/)
+or for multi tenant openshift-sdn.
+
+### Proxy
+
+If there are proxy settings for the broker, they will be forwarded to the pod
+by setting environment variables for ``HTTP_PROXY``, ``HTTPS_PROXY``, and
+``NO_PROXY``. 
+
+### Service Account
+
+Details of a kubernetes service account are mounted at
+``/var/run/secrets/kubernetes.io/serviceaccount``
+
+Keys contained in the Secret:
+
+* ca.crt: a base64-encoded CA certificate that was used to sign the certificate
```

Comment:
My understanding is that the `service-ca.crt is used to sign the certificate used by the broker to server its API as well as etcds API

Reply:
You appear to be right based on this: https://github.com/openshift/ansible-service-broker/blob/master/scripts/run_latest_build.sh#L89

I'm not sure what to do with this extra ``ca.crt`` that we don't know the purpose of.

Reply:
This is valuable to document, so not questioning its place in here, but do for my own edification, do we have a use-case for APB authors to use these files directly? The majority of the time, they just serve to authenticate the SA to do the APB work, no?

Reply:
There really isn't reason for a Service Bundle developer (APBs included as an instance of a Service Bundle) to interact with these directly. We may just want to link to the k8s documentation on what happens to service accounts in pods.

Reply:
Does the broker mount this information to this location? Or is it something kubernetes is doing automatically for some reason?